src/main/java/com/artipie/scripting/ScriptRunner.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/artipie/LICENSE.txt
+ */
+package com.artipie.scripting;
+
+import com.artipie.asto.Key;
+import com.artipie.asto.blocking.BlockingStorage;
+import com.artipie.settings.Settings;
+import com.artipie.settings.repo.RepositoriesFromStorage;
+import com.jcabi.log.Logger;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import javax.script.ScriptException;
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+
+/**
+ * Script runner.
+ * Job for running script in quartz
+ *
+ * @since 0.30
+ */
+public final class ScriptRunner implements Job {
+    @Override
+    public void execute(final JobExecutionContext context) throws JobExecutionException {
+        final Settings settings = (Settings) context.getJobDetail().getJobDataMap().get("settings");
+        final Key key = new Key.From(context.getJobDetail().getJobDataMap().getString("key"));
+        final BlockingStorage storage = new BlockingStorage(settings.configStorage());
```

Comment:
all these parameters from job data map can potentially be nulls. If any of them is null, it would be better to stop the job. Also, it's possible to make them fields and add setters, then quartz will set the values, check https://github.com/artipie/http/pull/551/files#diff-cede6ef368fefbd7e0f78b524ca0ea8109fb5a738824c4f2762c80a4e4f9f707 as an example